### PR TITLE
Add Generic From String Method

### PR DESCRIPTION
**Pull request by burakJs**
- Added `T? toPrimitiveFromGeneric<T>` method to String extension
- Added `string_constants` for storing string values of true and false
- Added `string_extension_mixin` for storing the utility method for converting from string to bool

### CODE REVIEW

File: lib/src/utility/mixin/string_extension_mixin.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  /// true or false constants, otherwise it returns null.
+  ///
+  /// Args:
+  ///   value (String): The `value` parameter is a string that represents a boolean value.
+  ///
+  /// Returns:
+  ///   The method `_getBoolFromString` returns a `bool?` value.
+  bool? _getBoolFromString(String value) {
+    final valueLowerCased = value.toLowerCase().trim();
+    final isEqualTrueString = valueLowerCased == StringConstants.trueConstant;
```
Comment (VB10) [inline lib/src/utility/mixin/string_extension_mixin.dart:14]: true.toString();

File: lib/src/string_extension.dart [hunk 2]
```diff
@@ -7,19 +7,38 @@ import 'package:flutter/material.dart';
...
 
+  /// The function `toPrimitiveFromGeneric` converts a generic value to a primitive type (bool, int,
+  /// double, or string) if possible.
+  ///
+  /// Returns:
+  ///   The method is returning a value of type T, which can be a boolean, integer, double, or string.
+  /// If none of these types match, it will return null.
+  T? toPrimitiveFromGeneric<T>() {
+    final value = _value;
+    if (value == null) return null;
```
Comment (VB10) [inline lib/src/string_extension.dart:34]: how can we manage the list ? you can send a exception when user want to cast it

**Review by VB10 — CHANGES_REQUESTED** [review]

File: lib/src/string_extension.dart [hunk 3]
```diff
@@ -7,19 +7,42 @@ import 'package:flutter/material.dart';
...
+  /// Throws a `ListTypeNotSupported` exception if the generic type is a list format.
+  ///
+  /// Type `T` is expected to be one of the supported primitive types: bool, int, double, String.
+  ///
+  /// Returns:
+  ///   The primitive value of type `T` or null.
+  T? toPrimitiveFromGeneric<T>() {
+    final value = _value;
+    if (value == null) return null;
+    if (checkListFormat(T.toString())) throw const ListTypeNotSupported();
```
Comment (VB10) [inline lib/src/string_extension.dart:38]: bu kod olmaz ya bool gelırse sonucta exceptıon atıyorsun adama

**Review by VB10 — CHANGES_REQUESTED** [review]

File: lib/src/string_extension.dart (hunk 3 above)
Comment (burakJs) [reply lib/src/string_extension.dart:38]: Hocam sadece list veya nested list oldugu zaman true diger turlu false dondugu icin sorun cikartmaz diye dusunmustum

Comment (burakJs) [reply lib/src/string_extension.dart:38]: image

**Review by VB10 — APPROVED** [review]

**Pull request merged**